docs/webmidi-mock.md

```diff
@@ -0,0 +1,177 @@
+| `id`           | The unique ID of the device associated with the port.           |
+| `name`         | The name of the device associated with the port.                |
+| `manufacturer` | The manufacturer of the device associated with the port.        |
+| `type`         | The type of port, either "input", or "output".                  |
+| `state`        | The state of the port, either "open", "pending", or "closed".   |
+| `connected`    | "connected" if the port is connected, "disconnected" otherwise. |
+
+All properties should be strings.  Although pattern matching is supported for other usages of "port specs", you should
+specify full and literal strings for each property used in the above component options.
+
+## Member Variables
+
+| Name                      | Description                                                                                   |
+|---------------------------|-----------------------------------------------------------------------------------------------|
+| `calls.requestMIDIAccess` | An array recording the full arguments passed in each call to `requestMIDIAccess` (see below). |
+| `accessEventTargets`      | An array containing the "event targets" (see below) for each access object.                   |
+| `inputs`                  | An object whose members are mocked MIDI input ports (see below).                              |
+| `outputs`                 | An object whose members are mocked MIDI output ports (see below).                             |
+
+### Function Call Registries
+
+Each mock instance includes a register of calls made to methods in its API. For example, from the
+perspective of an output port mock, `{port}.calls.sent.length` represents the number of times the `sent` method
+was called, and `{port}.calls.sent[0]` represents the arguments passed for the first call to the `sent` method.
+
+From the perspective of the enclosing `youme.tests.webMidiMock` component, the same information would be found at
+`{that}.outputs.get({id}).calls.sent.length` and `{that}.outputs.get({id}).calls.sent[0]`.
+
+### Event Targets
+
+In order to behave as the WebMIDI API does, we need to be able to trigger and respond to various events.  We use "event
+targets" to mock this behaviour, and these are exposed as member variables so that you can listen for or trigger events
+as part of your tests.
+
+### Mocks
+
+#### `MIDIAccess` Mock
+
+| Name               | Type          | Description                                                                              |
+|--------------------|---------------|------------------------------------------------------------------------------------------|
+| `sysexEnabled`     | `Boolean`     | Whether we can send/receive sysex messages using this access object.                     |
+| `inputs`           | `Map`         | A map of input mocks (see below).                                                        |
```

Comment:
Perhaps some more specific type here, a la @typedef that we sometimes get in JSDocs? E.g. Map<String, Something> ?

Reply:
Fixed.